bigtable/use-cases/fraudDetection/src/main/java/bigtable/fraud/beam/FraudDetection.java

```diff
@@ -0,0 +1,290 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package bigtable.fraud.beam;
+
+import bigtable.fraud.beam.utils.AggregatedData;
+import bigtable.fraud.beam.utils.CustomerDemographics;
+import bigtable.fraud.beam.utils.RowDetails;
+import bigtable.fraud.beam.utils.TransactionDetails;
+import bigtable.fraud.beam.utils.WriteCBTHelper;
+import com.google.cloud.aiplatform.v1.EndpointName;
+import com.google.cloud.aiplatform.v1.PredictRequest;
+import com.google.cloud.aiplatform.v1.PredictResponse;
+import com.google.cloud.aiplatform.v1.PredictionServiceClient;
+import com.google.cloud.aiplatform.v1.PredictionServiceSettings;
+import com.google.cloud.bigtable.beam.AbstractCloudBigtableTableDoFn;
+import com.google.cloud.bigtable.beam.CloudBigtableConfiguration;
+import com.google.cloud.bigtable.beam.CloudBigtableIO;
+import com.google.cloud.bigtable.beam.CloudBigtableScanConfiguration;
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ListValue;
+import com.google.protobuf.Value;
+import com.google.protobuf.util.JsonFormat;
+import java.io.IOException;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubIO;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.apache.hadoop.hbase.client.Scan;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class FraudDetection {
+  private FraudDetection() {
+  }
+
+  /**
+   * a Logger object to help logging details.
+   */
+  private static final Logger LOGGER = LoggerFactory.getLogger(
+      FraudDetection.class);
+
+  /**
+   * Set the field isFraud to true if the fraud_probability was >= 0.1 This is a
+   * configurable number that should be tuned depending on the ML model.
+   */
+  private static final double FRAUD_PROBABILITY_THRESHOLD = 0.1d;
+
+  /**
+   * Convert the line read from Cloud Pubsub into a TransactionDetails object.
+   */
+  static final DoFn<String, TransactionDetails> PREPROCESS_INPUT =
+      new DoFn<String, TransactionDetails>() {
+        @ProcessElement
+        public void processElement(
+            final DoFn<String, TransactionDetails>.ProcessContext c) {
+          try {
+            TransactionDetails transactionDetails = new TransactionDetails(
+                c.element());
+            c.output(transactionDetails);
+          } catch (Exception e) {
+            LOGGER.error("Failed to preprocess {}", c.element(), e);
+          }
+        }
+      };
+
+  // Read the transaction history for that customer, and outputs an
```

Comment:
nit: verb tenses should be consistent throughout; consider updating to "Reads" and keep an eye on it throughout